File: server/src/main/kotlin/domain/repository/PortfolioRepository.kt
package domain.repository

import domain.model.Portfolio

interface PortfolioRepository {
    suspend fun createPortfolio(
        creatorId: String,
        name: String,
        description: String,
        category: List<String>,
        photos: List<Pair<String, ByteArray>>
    ): String
    suspend fun getPortfolioById(portfolioId: String): Portfolio
    suspend fun getPortfoliosByCreatorId(creatorId: String): List<Portfolio>
    suspend fun searchPortfolios(
        categories: List<String>? = null,
        portfolioIds: List<String>? = null,
        sortBy: String? = null
    ): List<Portfolio>
    suspend fun updatePortfolio(
        portfolioId: String,
        name: String,
        description: String,
        category: List<String>,
        photoURLs: List<String>
    ): Portfolio
    suspend fun deletePortfolio(portfolioId: String): Boolean
}

File: ios/fotofolio2pto/DomainLayer/Repositories/PortfolioRepository.swift
//
//  PortfolioRepository.swift
//  fotofolio2pto
//
//  Created by Kryštof Příhoda on 24.06.2024.
//

import Foundation
import UIKit

public protocol PortfolioRepository {
    func createPortfolio(creatorId: String, name: String, photos: [IImage], price: Price, description: String, category: [String]) async throws
    func addToFlagged(portfolioId: String)
    func readAll(categories: [String]?, sortBy: SortByEnum?) async throws -> [Portfolio]
    func readFlagged() async throws -> [Portfolio]
    func readFlaggedIds() -> [String]
    func readImageFromURL(url: String) async throws -> UIImage
    func updatePortfolio(id: String, name: String, photos: [String], price: Price, description: String, category: [String]) async throws -> Portfolio
    func removeFromFlagged(portfolioId: String)
    func removeAllFlagged()
    func removePortfolio(id: String) async throws
}
